main.py

```diff
@@ -60,51 +70,84 @@ def scrape_table(headers):
     condition = "return document.getElementById('investments-table-object_last').classList.contains('disabled')"
     browser_lib.wait_for_condition(condition=condition, timeout=timedelta(seconds=20))
 
-    table = browser_lib.find_element('//*[@id="investments-table-object"]/tbody')
-    links = [link.get_attribute("href") for link in table.find_elements_by_tag_name("a")]
+    table = browser_lib.find_element('id:investments-table-object >> tag:tbody')
     table_data = [headers]
+    links = []
+    equalities = {}  # dict with "UII": "Name of investment"
 
     for row in table.find_elements_by_tag_name("tr"):
         cols = row.find_elements_by_tag_name("td")
         data_set = []
+
         for data in cols:
             data_set.append(data.text)
+            a_element = data.find_elements_by_tag_name("a")
+            if a_element:
+                links.append(a_element[0].get_attribute("href"))
+
         table_data.append(data_set)
+        equalities[data_set[0]] = data_set[2]
 
-    write_to_excel(sheetname="Individual Investments", bookname="spending.xlsx", content=table_data)
+    write_to_excel(sheetname="Individual Investments", bookname="spending.xlsx", content=table_data, method="append")
 
     for link in links:
-        browser_lib.go_to(link)
-        element = '//*[@id="business-case-pdf"]/a'
-        browser_lib.wait_until_element_is_visible(element, timeout=timedelta(seconds=10))
-        browser_lib.click_element(element)
-        browser_lib.assign_id_to_element(locator=element, id="USERFLAG")
-        condition = "return document.getElementById('USERFLAG').getAttribute('aria-busy') == \"false\""
-        browser_lib.wait_for_condition(condition=condition, timeout=timedelta(seconds=10))
+        file = download_file(link)
+        compare_values(equalities, file)
```

Comment:
So u compare **values from a pdf** with **all data from the table**, but you need to compare one to one (one pdf to one row from the table)

Reply:
Dictionary equalities contains pairs where UII is key and investment is value. In other words, each row of table is represented in dictionary like key: value pair. Once pdf is downloaded, UII and investment from THIS pdf are beeing compared with value from dictionary with UII as a unique key

Reply:
UII is a unique value, it can't be presented in two different rows (in this case)